internal/service/registry_service.go

```diff
@@ -154,3 +165,28 @@ func (s *registryServiceImpl) Publish(req model.PublishRequest) (*model.ServerRe
 	response := serverRecord.ToServerResponse()
 	return &response, nil
 }
+
+// validateNoDuplicateRemoteURLs checks that no other server is using the same remote URLs
+func (s *registryServiceImpl) validateNoDuplicateRemoteURLs(ctx context.Context, serverDetail model.ServerDetail) error {
+	// Check each remote URL in the new server for conflicts
+	for _, remote := range serverDetail.Remotes {
+		// Use filter to find servers with this remote URL
+		filter := map[string]any{
+			"remote_url": remote.URL,
```

Comment:
Would it make sense to add name to the filter here?
```suggestion
			"remote_url": remote.URL,
			"name": serverDetail.Name
```

Just trying to understand if we can move the "Check if any conflicting server has a different name" to the db query level and if it makes sense.